streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -737,13 +739,20 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
         // add configs required for stream partition assignor
         consumerProps.put(REPLICATION_FACTOR_CONFIG, getInt(REPLICATION_FACTOR_CONFIG));
+        consumerProps.put(APPLICATION_SERVER_CONFIG, getString(APPLICATION_SERVER_CONFIG));
         consumerProps.put(NUM_STANDBY_REPLICAS_CONFIG, getInt(NUM_STANDBY_REPLICAS_CONFIG));
         consumerProps.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, StreamPartitionAssignor.class.getName());
         consumerProps.put(WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG, getLong(WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG));
-        consumerProps.put(APPLICATION_SERVER_CONFIG, getString(APPLICATION_SERVER_CONFIG));
+
+        // add admin retries configs for creating topics
         final AdminClientConfig config = new AdminClientConfig(getClientPropsWithPrefix(ADMIN_CLIENT_PREFIX, AdminClientConfig.configNames()));
         consumerProps.put(adminClientPrefix(AdminClientConfig.RETRIES_CONFIG), config.getInt(AdminClientConfig.RETRIES_CONFIG));
 
+        // add admin and topic configs required for creating topics, also add batch size of producer for verification
+        final Map<String, Object> producerProps = getClientPropsWithPrefix(PRODUCER_PREFIX, ProducerConfig.configNames());
+        consumerProps.put(producerPrefix(ProducerConfig.BATCH_SIZE_CONFIG), producerProps.containsKey(ProducerConfig.BATCH_SIZE_CONFIG) ? (Integer) producerProps.get(ProducerConfig.BATCH_SIZE_CONFIG) : 16384);
```

Comment:
Where does `16384` come from? Sounds like the producer default config value. Shouldn't we get this out of the `ProducerConfig` similar to `retires` from `AdminClientConfig` ?

Reply:
Yes it is indeed from `ProducerConfig`, unfortunately this value is not exposed from `ProducerConfig` object and hence cannot be leveraged.

Reply:
Yes. We have the same issue with `AdminClientConfig` and `retries` -- thus, we instantiate a `AdminClientConfig` got get the default out of it. Can we do the same thing here and instantiate a `ProducerConfig` object?

I now it's not very nice code, but still better than hardcoding the value.